src/lib/UniversalDiscovery/Event/Subscriber/SectionAssign.php

```diff
@@ -97,20 +96,15 @@ private function getRestrictedContentTypes(array $hasAccess): array
         }
 
         if (empty($restrictedContentTypesIds)) {
-            return $restrictedContentTypes;
+            return [];
         }
 
-        $restrictedContentTypesIds = array_unique(array_merge(...$restrictedContentTypesIds));
```

Comment:
Review Note: I didn't see why array_merge() was used here so removed it in the new code,